api/main.go

```diff
@@ -84,6 +83,19 @@ func main() {
 
 	publicAPI := e.Group("/api")
 	internalAPI := e.Group("/internal")
+	internalAPI.GET("/devices/:uid/offline", func(c echo.Context) error {
```

Comment:
Using GET or HEAD on a resource URL, should NEVER change the resource. Please change it to POST